File: rzops-web/src/lib/types/provider.ts
export interface ProviderResponse {
  id: string;
  name: string;
  provider_types: string[];
  contact_name: string | null;
  contact_phone: string | null;
  contact_qq: string | null;
  fax: string | null;
  address: string | null;
  website: string | null;
  description: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface ProviderListResponse {
  data: ProviderResponse[];
  count: number;
}

export interface CreateProviderRequest {
  name: string;
  provider_types?: string[];
  contact_name?: string;
  contact_phone?: string;
  contact_qq?: string;
  fax?: string;
  address?: string;
  website?: string;
  description?: string;
  status?: string;
}

export interface UpdateProviderRequest extends Partial<CreateProviderRequest> {}

export interface ListProvidersQuery {
  q?: string;
  status?: string;
  page?: number;
  per_page?: number;
}

File: rzops-api/domain/src/models/provider.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;


/// Provider entity — maps to cmdb_provider.
#[derive(Debug, Clone)]
pub struct Provider {
    pub id: Uuid,
    pub name: String,
    pub provider_types: Vec<String>, // stored as JSONB array of ProviderType codes
    pub contact_name: Option<String>,
    pub contact_phone: Option<String>,
    pub contact_qq: Option<String>,
    pub fax: Option<String>,
    pub address: Option<String>,
    pub website: Option<String>,
    pub country: Option<String>,
    pub description: Option<String>,
    pub status: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
